tests/octf/analytics/statistics/WorksetCalculatorTest.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright(c) 2012-2018 Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause-Clear
+ */
+
+#include <gtest/gtest.h>
+#include <limits>
+#include <random>
+#include <vector>
+#include <octf/analytics/statistics/WorksetCalculator.h>
+
+using namespace octf;
+
+TEST(WorksetCalculator, exclusiveRanges) {
+    WorksetCalculator wc;
+
+    int ranges = 100000;
+    int range = 0;
+
+    int len = 10;
+    int begin = 0;
+
+    // Add exclusive ranges like so:
+    // [0-10], [20-30], [40-50], ...
+    for (; range < ranges; begin += len + 10, range++) {
+        wc.insertRange(begin, len);
+    }
+
+    ASSERT_EQ(wc.getWorkset(), ranges * len);
+}
+
+TEST(WorksetCalculator, identicalRanges) {
+    WorksetCalculator wc;
+
+    int ranges = 100000;
+    int range = 0;
+
+    int len = 10;
+    int begin = 0;
+
+    for (; range < ranges; range++) {
+        wc.insertRange(begin, len);
+    }
+
+    ASSERT_EQ(wc.getWorkset(), len);
+}
+
+TEST(WorksetCalculator, subRanges) {
+    WorksetCalculator wc;
+
+    int ranges = 100000;
+    int range = 0;
+
+    int maxLen = ranges * 2 + 2;
+    int currentLen = maxLen;
+    int currentBegin = 0;
+
+    // Insert the biggest range
+    wc.insertRange(currentBegin, maxLen);
+
+    // Add subranges like so:
+    // First range: [0-100]
+    // Next ranges: [1-99], [2-98], ...
+    for (; range < ranges; range++) {
+        currentBegin++;
+        currentLen--;
+        assert(currentLen > 0);
```

Comment:
Maybe we should use gtest assert

Reply:
I think it is an assert on the test code and not on the tested code and should be left as it is.